src/hooks/users.ts

```diff
@@ -105,6 +108,40 @@ export const useRecentWorldUsers = (): {
   );
 };
 
+type LastSeenAt = {
+  venueName: string;
+  venueUrl: string;
+};
+
+export const useRecentWorldUser = (
+  userId?: string
+): {
+  user?: WithId<User>;
+  lastSeenAt: LastSeenAt;
+} => {
+  const { recentWorldUsers } = useRecentWorldUsers();
+
+  const user = useMemo(() => {
+    return recentWorldUsers.find((worldUser) => worldUser.id === userId);
+  }, [userId, recentWorldUsers]);
+
+  const venueName = getUserCurrentLocation(user) ?? "";
+  const venueUrl = createUrlSafeName(venueName);
```

Comment:
What does `createUrlSafeName` do? Not sure if this is going to be correct. `venueName` is something entirely different to `venueId`. The URLs use `venueId`.

Reply:
Actually, I could extract only `venueName` from user

```
const venueName = Object.keys(recentUser.lastSeenIn)[0] ?? "";
```
And `createUrlSafeName` method is the right (IMHO) way to get venueUrl. It's used to create venueUrl from venueName in the relevant form